### PR TITLE
[fix](json) fix json int128 overflow

### PR DESCRIPTION
**Pull request by xiaokang**
## Proposed changes

Issue Number: close #xxx

Save uint64 to int128 value in json binary format to fix the following overflow error:

```
W0811 14:25:10.943372 2487636 jsonb_parser_simd.h:301] overflow number: 9883712598971902813
W0811 14:25:10.943392 2487636 jsonb_parser_simd.h:214] parse object value failed
W0811 14:25:10.944248 2487636 jsonb_parser_simd.h:301] overflow number: 14632648102329842500
```

NOTE: simdjson lib does not support int128, so this pr just resolve problem for uint64 up to 18446744073709551615. If the number is larger than 18446744073709551615, it still cause overflow problem.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/jsonb_stream.h [hunk 1]
```diff
@@ -126,9 +130,40 @@ class JsonbOutStream : public std::ostream {
         size_ += len;
     }
 
-    void write(__int128 l) {
-        // TODO
-        assert(false);
+    // TODO: use more efficient impl for int128
+    void write(int128_t l) {
```
Comment (eldenmoon) [inline be/src/util/jsonb_stream.h:134]: maybe we could use
```
fmt::memory_buffer buffer;
fmt::format_to(buffer, "{}", l);
```

File: be/src/util/jsonb_stream.h [hunk 2]
```diff
@@ -126,9 +130,40 @@ class JsonbOutStream : public std::ostream {
...
+
+        // snprintf from low digit to high digit
+        while (l != 0) {
+            // get lowest digit
+            int n = l % 10;
+            if (n < 0) {
+                n = 0 - n;
+            }
+            assert(n >= 0 && n < 10);
+            len = snprintf(head_ + size_, MAX_INT128_DIGITS, "%d", n);
```
Comment (eldenmoon) [inline be/src/util/jsonb_stream.h:157]: this is slow

File: be/src/util/jsonb_stream.h (hunk 1 above)
Comment (xiaokang) [reply be/src/util/jsonb_stream.h:134]: changed to `fmt::format_to_n(head_ + size_, MAX_INT128_DIGITS, "{}", l)`

File: be/src/util/jsonb_parser_simd.h [hunk 3]
```diff
@@ -296,21 +298,27 @@ class JsonbParserTSIMD {
                 return;
             }
         } else if (num.is_int64() || num.is_uint64()) {
-            if (num.is_uint64() && num.get_uint64() > std::numeric_limits<int64_t>::max()) {
+            if (num.is_uint64() && num.get_uint64() > std::numeric_limits<int128_t>::max()) {
```
Comment (eldenmoon) [inline be/src/util/jsonb_parser_simd.h:301]: get_uint64 will return uint64_t is will nerver larger than `std::numeric_limits<int128_t>::max()`

Comment (xiaokang) [reply be/src/util/jsonb_parser_simd.h]: deleted

**Comment by xiaokang** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.53 seconds
 stream load tsv:          554 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162225270 Bytes

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by xiaokang** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by csun5285 — COMMENTED** [review]
LGTM

**Review by csun5285 — APPROVED** [review]
LGTM

**Pull request merged**